hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/StoreSnapshotTasklet.java

```diff
@@ -165,4 +184,27 @@ public String toString() {
         /** Input is done, terminal state. */
         DONE
     }
+
+    private static class Metrics implements DynamicMetricsProvider {
+
+        private long bytes;
+        private long keys;
+        private long chunks;
+
+        synchronized void set(long bytes, long keys, long chunks) {
```

Comment:
why use `synchronized` here?

Reply:
The intention is to update/query the 3 values in an atomic fashion, ie. never publish metrics in which for example the byte count is from one snapshot and the keys from another. But not sure if these benefits outweigh the costs (costs which in turn shouldn't be high, since this is not really a contended lock).